docs/tutorials/tfx/stub_template.md

```diff
@@ -0,0 +1,108 @@
+
+1.  In `kubeflow_dag_runner.py`, add `StubComponentLauncher` class to `KubeflowDagRunnerConfig`'s `supported_launcher_class` to enable launch of stub executors:
+
+``` python
+  runner_config = kubeflow_dag_runner.KubeflowDagRunnerConfig(
+      # TODO(StubExecutor): Uncomment below to use stub executors.
+      supported_launcher_classes=[
+        stub_component_launcher.StubComponentLauncher
+      ],
+```
+
+2.  In `launcher/stub_component_launcher.py`, modify the default list of component ids that are to be replaced with BaseStubExecutor. If you want to replace component executor with a custom stub executor, which inherits BaseStubExecutor, modify stubbed_component_map to include component id to custom stub executor mapping. If 
+```python
+# GCS directory where KFP outputs are recorded
+test_data_dir = "gs://{}/testdata".format(configs.GCS_BUCKET_NAME)
+# TODO(StubExecutor): customize self.stubbed_component_ids to replace components
+# with BaseStubExecutor
+stubbed_component_ids = ['CsvExampleGen', 'StatisticsGen',
+                         'SchemaGen', 'ExampleValidator',
+                         'Trainer', 'Transform', 'Evaluator', 'Pusher']
+# TODO(StubExecutor): (Optional) Use stubbed_component_map to insert custom stub
+# executor class as a value and component id as a key.
+stubbed_component_map = {}
+```
+
+#### In Jupyter lab file editor:
+>**Double-click to open `kubeflow_dag_runner.py`**. 
+Uncomment `supported_launcher_classes` argument of `KubeflowDagRunnerConfig` to be able to launch stub executors (Tip: search for comments containing `TODO(StubExecutor):`),  Make sure to save `kubeflow_dag_runner.py` after you edit it.
+
+
+
+>**Double-click to change directory to `launcher` and double-click again to open `stub_component_launcher.py`.**
+Make sure to set `test_data_dir` to the GCS directory where KFP outputs are recorded, or `output_dir` (Tip: search for comments containing `TODO(StubExecutor):`). Customize the list `self.stubbed_component_ids`, or ids of components that should be replaced with BaseStubExecutor(Tip: search for comments containing `TODO(StubExecutor):`). You may additionally insert custom stub executor in `self.stubbed_component_map` with component id as a key and custom stub executor class as value (Tip: search for comments containing `TODO(StubExecutor):`). Make sure to save `stub_component_launcher.py` after you edit it.
+
+### 3. Update and run the pipeline with stub executors
+Update the existing pipeline with modified pipeline definition with stub executors.
+```bash
+tfx pipeline update
+--pipeline-path=kubeflow_dag_runner.py
+--endpoint={ENDPOINT} --engine=kubeflow
+```
+
+Run the following command to create a new execution run of your updated pipeline.
+
+```bash
+tfx run create --pipeline-name {PIPELINE_NAME} --endpoint={ENDPOINT} --engine=kubeflow
+```
```

Comment:
Please make sure explaining clean-up step as well, for example, killing port-forwarding process.